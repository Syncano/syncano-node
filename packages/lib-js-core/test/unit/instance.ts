import * as chai from 'chai'
import * as chaiAsPromised from 'chai-as-promised'
import * as nock from 'nock'
import Server from '../../src'

chai.use(chaiAsPromised)
chai.should()

describe('Instance', () => {
  const instanceName = 'testInstance'
  let api
  let instance

  beforeEach(() => {
    const server = new Server({
      token: 'testKey',
      instanceName
    })
    instance = server.instance
    api = nock(`https://${process.env.SYNCANO_HOST || 'api.syncano.io'}`)
  })

  describe('#create', () => {
    it('should create syncano instance', () => {
      api.post(`/v2/instances/`, {name: 'my-insta'}).reply(200)

      return instance.create({name: 'my-insta'}).should.be.fulfilled
    })
  })

  describe('#delete', () => {
    it('should delete syncano instance', () => {
      api.delete(`/v2/instances/my-insta/`).reply(200)

      return instance.delete('my-insta').should.be.fulfilled
    })
  })
})
