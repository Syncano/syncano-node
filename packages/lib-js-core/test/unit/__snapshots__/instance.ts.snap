// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instance #create should create syncano instance 1`] = `
Object {
  "name": "my-insta",
}
`;

exports[`Instance #list should list syncano instances 1`] = `
Array [
  Object {
    "name": "instance1",
  },
  Object {
    "name": "instance2",
  },
]
`;
